docs: update plugin guidelines

bastimeyer opened the pull request:
The last revision of the plugin request/submission guidelines was made two years ago in #1838.
Lately we've seen a few requests and PRs for radio stations, which we don't want to merge. This should be included in the list of plugin restrictions.

I've also added a generic pull request template, since we don't have one. Nothing big, just a small reminder to read the guidelines, add a comment and run local checks if possible.

gravyboat commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,12 @@
+
+If possible, run the tests, code linting tools and documentation builds locally on your system first to avoid unnecessary build failure surprises.
+
+Also don't forget to add a meaningful description of your changes, so that the reviewing process is as simple as possible for the maintainers.
```
Should we add a link to an example of a good PR here?

bastimeyer replied:
No, I'll add something to the docs which can then be linked. This will partially solve #3195

gravyboat commented on CONTRIBUTING.md:
```diff
@@ -47,7 +47,10 @@ Plugins which fall under the following categories will not be implemented or con
+8. Sites which are unmaintained
+
+9. Sites which don't provide any video streaming content, eg. radio stations
+
```
I think it's worth adding a few additional lines such as:
```
10. Sites which create a live stream of previously recorded content, such as TV shows, movies, news broadcasts, etc.

11. Sites which have no way to determine viewership numbers

12. Sites which are static cameras of a physical location

13. Sites which are in beta or undergoing heavy amounts of development and may change rapidly
```

This should cover us for more plugins and keep people from wasting their time and ours.

bastimeyer replied:
I agree with all of those, but I feel like a list of 13 points makes it a bit too much IMO.

gravyboat replied:
Agreed, your changes are good.

gravyboat reviewed (changes requested):
I added a few changes for review and discussion, let me know what you think.

bastimeyer commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,12 @@
+Before you continue, please make sure that you have read and understood the contribution guidelines, otherwise your changes may be rejected:
+https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#contributing-to-streamlink
+
+If possible, run the tests, code linting tools and documentation builds locally on your system first to avoid unnecessary build failure surprises.
```
My goal was to have this sentence as short as possible. "code linting tools" and "documentation builds" were related to "run locally". I was thinking about listing the tasks, but that made the template a bit too long, and I want a concise one.

> do we even have information that is easy to find on what commands need to be run to perform these actions?

No, and there is also no documentation on how to set up a dev environment. I can add this to the docs as a new submenu before the pull request template gets merged. Or keep the template out of this PR and add it with the docs upgrade. I think this is a better idea.

gravyboat replied:
Sounds good. Adding with the docs upgrade makes more sense.

bastimeyer commented:
- merged "mostly hosting VODs" with "live stream of previously recorded content"
- merged "unmaintained" with "in beta or are undergoing heavy amounts of development and may change rapidly"
- added "no way to determine viewership numbers"
- added "static cameras of a physical location"
- re-ordered the list a bit

Is that okay?

gravyboat commented:
Yep! Merging.
